src/main/java/org/spongepowered/common/command/brigadier/tree/SpongeArgumentCommandNode.java

```diff
@@ -201,4 +206,14 @@ public final void parse(final StringReader reader, final CommandContextBuilder<C
         return this.parser.getExamples();
     }
 
+    @Override
+    public void addChild(final CommandNode<CommandSource> node) {
+        super.addChild(node);
+        this.childNodes.add(node);
+    }
+
+    @Override
+    public Collection<CommandNode<CommandSource>> getUnsortedChildren() {
+        return this.childNodes;
```

Comment:
This is returning a mutable collection but `SpongeRootCommandNode` is returning an immutable one. Is it intended ?

Reply:
The intention was to just return the mutable collection for the simple reason that it's only being used in an iteration in one of the mixins to therefore avoid a copy of what could end up being a large collection. I may wrap it in `Collections.unmodifiableCollection(...)` or something.

As for the root collection, probably me in auto-mode.